### ISSUE TITLE
[Question]:  Ragflow docker image without embedding models not loading on MacBook M1Max

### ISSUE THREAD

**Issue by embedsri** [infiniflow/ragflow#3228]
### Describe your problem

I'm unable to load this image that I built without embedding models on my macbook.
docker image ls                                                 
REPOSITORY                                      TAG                            IMAGE ID       CREATED         SIZE
infiniflow/ragflow                              dev-slim                       49bcb5b26952   2 hours ago     7.04GB

suc@Srirams-MacBook-Pro ragflow % docker run 49bcb5b26952          
Traceback (most recent call last):
  File "/ragflow/api/utils/file_utils.py", line 126, in load_yaml_conf
    with open(conf_path) as f:
         ^^^^^^^^^^^^^^^
FileNotFoundError: [Errno 2] No such file or directory: '/ragflow/conf/service_conf.yaml'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/ragflow/api/ragflow_server.py", line 26, in <module>
    from api.apps import app
  File "/ragflow/api/apps/__init__.py", line 26, in <module>
    from api.db.db_models import close_connection
  File "/ragflow/api/db/db_models.py", line 33, in <module>
    from api.settings import DATABASE, stat_logger, SECRET_KEY, DATABASE_TYPE
  File "/ragflow/api/settings.py", line 19, in <module>
    from api.utils.file_utils import get_project_base_directory
  File "/ragflow/api/utils/__init__.py", line 67, in <module>
    use_deserialize_safe_module = get_base_config(
                                  ^^^^^^^^^^^^^^^^
  File "/ragflow/api/utils/__init__.py", line 58, in get_base_config
    config = file_utils.load_yaml_conf(config_path)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/ragflow/api/utils/file_utils.py", line 130, in load_yaml_conf
    raise EnvironmentError(
OSError: [Errno loading yaml file config from /ragflow/conf/service_conf.yaml failed:] [Errno 2] No such file or directory: '/ragflow/conf/service_conf.yaml'
Traceback (most recent call last):
  File "/ragflow/api/utils/file_utils.py", line 126, in load_yaml_conf
    with open(conf_path) as f:
         ^^^^^^^^^^^^^^^
FileNotFoundError: [Errno 2] No such file or directory: '/ragflow/conf/service_conf.yaml'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/ragflow/api/ragflow_server.py", line 26, in <module>
    from api.apps import app
  File "/ragflow/api/apps/__init__.py", line 26, in <module>
    from api.db.db_models import close_connection
  File "/ragflow/api/db/db_models.py", line 33, in <module>
    from api.settings import DATABASE, stat_logger, SECRET_KEY, DATABASE_TYPE
  File "/ragflow/api/settings.py", line 19, in <module>
    from api.utils.file_utils import get_project_base_directory
  File "/ragflow/api/utils/__init__.py", line 67, in <module>
    use_deserialize_safe_module = get_base_config(
                                  ^^^^^^^^^^^^^^^^
  File "/ragflow/api/utils/__init__.py", line 58, in get_base_config
    config = file_utils.load_yaml_conf(config_path)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/ragflow/api/utils/file_utils.py", line 130, in load_yaml_conf
    raise EnvironmentError(
OSError: [Errno loading yaml file config from /ragflow/conf/service_conf.yaml failed:] [Errno 2] No such file or directory: '/ragflow/conf/service_conf.yaml'
Traceback (most recent call last):
  File "/ragflow/api/utils/file_utils.py", line 126, in load_yaml_conf
    with open(conf_path) as f:
         ^^^^^^^^^^^^^^^
FileNotFoundError: [Errno 2] No such file or directory: '/ragflow/conf/service_conf.yaml'

**Comment by KevinHuSh** [comment]
Copy docker/service_conf.yaml intto container at /ragflow/conf/

**Comment by KevinHuSh** [comment]
The ES seems down. It's complicated for Macs.

**Comment by embedsri** [comment]
@KevinHuSh Thank you. What's the ES in Ragflow context?  It appears to be the MongoDB elasticsearch vector database. How to resolve this error so I have ragflow working on mac?

**Comment by gschmutz** [comment]
@embedsri  it works on my machine, Apple M1 Pro. Why don't you use the docker compose stack provided in the `docker` folder?

```
$ cd ragflow/docker
$ docker compose -f docker-compose.yml up -d
```

**Comment by embedsri** [comment]
@gschmutz Thanks. I just wanted to build one from source but took your suggestion and created the docker image. But I'm unable to add files to the knowledge base because I see this warning which doesn't allow me to do so:
<img width="1134" alt="Screenshot 2024-11-11 at 8 19 41 PM" src="https://github.com/user-attachments/assets/466890ef-14d1-43f8-9fa7-fce1a2604eaf">

**Comment by KevinHuSh** [comment]
Follow the warning message. At least you need to set an embedding model.

**Comment by embedsri** [comment]
@KevinHuSh  Where in the RAGFlow interface is 'Settings>Model Providers'?
 I've set the embedding model in  'System Model Settings' 
<img width="1134" alt="Screenshot 2024-11-12 at 9 15 40 PM" src="https://github.com/user-attachments/assets/28721586-1469-4673-a283-6ea12be21a32">
Am I just something simple and obvious?

**Comment by KevinHuSh** [comment]
I'm not sure if yuo setup the chat LLM?

**Comment by embedsri** [comment]
Is the knowledgebase not stored in persistent storage? After seeing some errors, I restarted the docker image but the knowledge base and the LLMs setup was lost. So, I retried this:
docker compose -f docker-compose.yml up -d                          
WARN[0000] The "HF_ENDPOINT" variable is not set. Defaulting to a blank string. 
WARN[0000] The "MACOS" variable is not set. Defaulting to a blank string. 
[+] Running 5/5
 ✔ Container ragflow-minio   Started                                                                                                                                                                             0.2s 
 ✘ Container ragflow-es-01   Error                                                                                                                                                                               7.7s 
 ✘ Container ragflow-mysql   Error                                                                                                                                                                               1.2s 
 ✔ Container ragflow-server  Created                                                                                                                                                                             0.0s 
 ✔ Container ragflow-redis   Started                                                                                                                                                                             0.2s 
dependency failed to start: container ragflow-mysql is unhealthy
and the rag flow-server process is not up. 
docker % docker container ls                       
CONTAINER ID   IMAGE                                                  COMMAND                  CREATED       STATUS                                     PORTS                              NAMES
479739b03fb7   mysql:8.0.39                                           "docker-entrypoint.s…"   11 days ago   Restarting (1) 2 seconds ago                                                  ragflow-mysql
b6399d81ec30   quay.io/minio/minio:RELEASE.2023-12-20T01-00-02Z       "/usr/bin/docker-ent…"   11 days ago   Up 11 seconds                              0.0.0.0:9000-9001->9000-9001/tcp   ragflow-minio
5b4ea05202d9   docker.elastic.co/elasticsearch/elasticsearch:8.11.3   "/bin/tini -- /usr/l…"   11 days ago   Up Less than a second (health: starting)   9300/tcp, 0.0.0.0:1200->9200/tcp   ragflow-es-01
c132b6fdd3fa   valkey/valkey:8                                        "docker-entrypoint.s…"   11 days ago   Up 11 seconds                              0.0.0.0:6379->6379/tcp             ragflow-redis

How do I resolve this?

**Comment by liguop2008** [comment]
我也出现这个问题了，有没有解决方案？大神，
docker compose -f docker-compose.yml up -d 这个一切正常
docker logs -f ragflow-server 提示错误，如下：
OSError: [Errno loading yaml file config from /ragflow/conf/service_conf.yaml failed:] [Errno 2] No such file or directory: '/ragflow/conf/service_conf.yaml'
service_conf.yaml是存在的。权限也是够的。

**Comment by mikejhan4455** [comment]
The main branch and the dev version image are under development and change every time. 

Checkout to the branch '0.13.0' and setting `RAGFLOW_IMAGE=infiniflow/ragflow:v0.13.0` solves the problem for me.

**Comment by liguop2008** [comment]
您的邮件已收到！祝工作顺利！

**Comment by KevinHuSh** [comment]
Pull the latest code.

**Comment by liguop2008** [comment]
> The main branch and the dev version image are under development and change every time.
> 
> Checkout to the branch '0.13.0' and setting `RAGFLOW_IMAGE=infiniflow/ragflow:v0.13.0` solves the problem for me.
我用的是这个RAGFLOW_IMAGE=infiniflow/ragflow:v0.13.0才提示上面的错误。

**Comment by embedsri** [comment]
Thank you for your assistance. I'm making progress but it is 2 steps forward and 1 step backward.
This is what I see for remote branches: 

MacBookPro ragflow % git branch -r -v
  origin/HEAD -> origin/main
  origin/api  1fafdb84 fix(API): fixed retrieval api parameters matching (#2550)
  origin/main cc5960b8 Vietnamese language support on web (#3549)

I'm on the main branch and when I just try to download a prebuilt Docker image for RAGFlow slim (dev-slim) with these in the .env
RAGFLOW_IMAGE=infiniflow/ragflow:v0.13.0-slim
MACOS=1:
 docker % docker compose -f docker-compose.yml up -d
[+] Running 5/6
 ⠹ Container ragflow-mysql                                                                                                                                Starting                                               2.6s 
 ✔ Container ragflow-minio                                                                                                                                Started                                                2.6s 
 ✔ Container ragflow-es-01                                                                                                                                Started                                                2.6s 
 ✔ Container ragflow-redis                                                                                                                                Started                                                2.6s 
 ✔ Container ragflow-server                                                                                                                               Recreated                                              0.1s 
 ! ragflow The requested image's platform (linux/amd64) does not match the detected host platform (linux/arm64/v8) and no specific platform was requested                                                        0.0s 
Error response from daemon: Mounts denied: 
The path /Users/suc/ragflow/docker/init.sql is not shared from the host and is not known to Docker.
You can configure shared paths from Docker -> Preferences... -> Resources -> File Sharing.
See https://docs.docker.com/desktop/settings/mac/#file-sharing for more info.

I'd loaded prebuilt docker image for M1Max arm64/v8 before as we all can see earlier in the thread. What happened to that now?

So, I'm back to  building an image for M1Max bash build_docker_image.sh slim which is also giving an error:

 => ERROR [production 19/23] COPY --from=builder /ragflow/web/dist /ragflow/web/dist                                                                                                                             0.0s 
------                                                                                                                                                                                                                
 > [production 19/23] COPY --from=builder /ragflow/web/dist /ragflow/web/dist:                                                                                                                                        
------                                                                                                                                                                                                                
Dockerfile.slim:126                                                                                                                                                                                                   
--------------------                                                                                                                                                                                                  
 124 |     
 125 |     # Copy compiled web pages
 126 | >>> COPY --from=builder /ragflow/web/dist /ragflow/web/dist
 127 |     
 128 |     # Copy Python environment and packages
--------------------
ERROR: failed to solve: failed to compute cache key: failed to calculate checksum of ref lzlmsnquca292lao183jmu95t::nqwa5uq9uyftv19kzbu4hic6f: "/ragflow/web/dist": not found

View build details: docker-desktop://dashboard/build/desktop-linux/desktop-linux/fntjcoufsdt2mqwe8b37dh6eq

**Comment by yuzhichang** [comment]
@embedsri [README](https://github.com/infiniflow/ragflow?tab=readme-ov-file#-build-a-docker-image-without-embedding-models)
